components/mailchimp/sources/link-clicked/link-clicked.js

```diff
@@ -0,0 +1,101 @@
+const common = require("../common/timer-based");
+
+module.exports = {
+  ...common,
+  key: "link-clicked",
+  name: "Link Clicked",
+  description:
+    "Emit new event when a recipient clicks a pre-specified link in an specific campaign.",
+  version: "0.0.1",
+  type: "source",
+  dedupe: "unique",
+  props: {
+    ...common.props,
+    campaignId: {
+      propDefinition: [
+        common.props.mailchimp,
+        "campaignId",
+      ],
+      description:
+        "The unique ID of the campaign you'd like to watch for new clicks on links.",
+    },
+    linkId: {
+      propDefinition: [
+        common.props.mailchimp,
+        "linkId",
+        (c) => ({
+          campaignId: c.campaignId,
+        }),
+      ],
+    },
+    uniqueClicksOnly: {
+      type: "boolean",
+      label: "Unique Clicks Only?",
+      description: "Whether to count every link click or only count clicks coming from each user only once",
+      default: false,
+    },
+  },
+  hooks: {
+    async deploy() {
+      // Emits sample events on the first run during deploy.
+      return await this.emitReportSampleEvents(this.campaignId, this.linkId, (new Date()).getTime());
+    },
```

Comment:
This `await` can be removed.
```suggestion
    async deploy() {
      // Emits sample events on the first run during deploy.
      return this.emitReportSampleEvents(this.campaignId, this.linkId, (Date.now());
    },
```